project_euler/problem_205/sol1.py

```diff
@@ -0,0 +1,94 @@
+"""
+Project Euler Problem 205: https://projecteuler.net/problem=205
+
+Peter has nine four-sided (pyramidal) dice, each with faces numbered 1, 2, 3, 4.
+Colin has six six-sided (cubic) dice, each with faces numbered 1, 2, 3, 4, 5, 6.
+Peter and Colin roll their dice and compare totals: the highest total wins. The result
+is a draw if the totals are equal.
+What is the probability that Pyramidal Pete beats Cubic Colin? Give your answer rounded
+to seven decimal places in the form 0.abcdefg
+
+In the solution for all possible permutations of each player's dice points the sum of
+points and number of occurrences are calculated. The number of occurrences are divided
+by number of dice draws to get their probability.
+Then for each permutation of the possible dice sum per draw between the two players the
+ probabilities each time where Peter wins are added up to calculate the solution.
+To generate the permutations 'itertools.product' is used.
+
+"""
+
+from itertools import product
+from typing import List
+
+
+def probability_of_sums(
+    dice_max_points: int = 6, dice_number: int = 6
+) -> (List[int], List[float]):
+    """
+    Returns the list of possible sums and their probabilities of dice_number dices with
+    numbers from dice_min to dice_max
+
+    >>> probability_of_sums(dice_max_points=5, dice_number=2)
+    ([2, 3, 4, 5, 6, 7, 8, 9, 10], \
+[0.04, 0.08, 0.12, 0.16, 0.2, 0.16, 0.12, 0.08, 0.04])
+
+    """
+
+    sums = []
+    counter = []
+    for dices in product(range(1, dice_max_points + 1), repeat=dice_number):
+        s = sum(dices)
+        if s not in sums:  # sum has not occurred, set count to 1
+            sums.append(s)
+            counter.append(1)
+        else:  # sum has occurred, increment sum by 1
+            idx = sums.index(s)
+            counter[idx] += 1
+    total = sum(counter)
+    probability = [_t / total for _t in counter]
```

Comment:
I don't think this is a private attribute: `_t`, please provide a better and descriptive name.